src/tests/ftest/erasurecode/restart.py

```diff
@@ -57,29 +58,38 @@ def execution(self, agg_check=None):
         # 4.  step-5 for Restart_before_agg test
         self.log_step("Enable aggregation")
         self.pool.set_property("reclaim", "time")
-        # Aggregation will start in 20 seconds after it sets to time mode. So wait for 20
-        # seconds and restart all the servers.
-        time.sleep(20)
 
         # 5.  step-6 for Restart_before_agg test
         self.log_step("Rerun IOR")
-        self.ior_write_dataset(operation="Auto_Write", percent=self.percent)
-        init_free_space = self.pool.get_total_free_space(refresh=True)
-        self.log.info("pool free space after aggregation started: %s", init_free_space)
+        self.ior_write_dataset(storage='SCM', operation="Auto_Write", percent=self.percent)
+        init_total_free_space = self.pool.get_total_free_space(refresh=True)
+        pool_info = self.pool.get_pool_daos_space()
+        init_scm_free_space = pool_info["s_free"][0]
+        init_nvme_free_space = pool_info["s_free"][1]
+        self.log.info("After aggregation started: total_free: %s, scm_free: %s, nvme_free: %s",
+                      init_total_free_space, init_scm_free_space, init_nvme_free_space)
 
         # 6.  step-7 for Restart_before_agg test
-        self.log_step("Verify aggregation triggered")
+        self.log_step("Verify aggregation triggered, scm free space move to nvme")
         aggregation_detected = False
         timed_out = False
         start_time = time.time()
         while not aggregation_detected and not timed_out:
-            current_free_space = self.pool.get_total_free_space(refresh=True)
-            self.log.info("pool free space during Aggregation = %s", current_free_space)
-            if current_free_space > init_free_space + aggr_threshold:
+            pool_info = self.pool.get_pool_daos_space()
+            current_scm_free_space = pool_info["s_free"][0]
+            current_nvme_free_space = pool_info["s_free"][1]
+            diff_scm_free_space = current_scm_free_space - init_scm_free_space
+            diff_nvme_free_space = init_nvme_free_space - current_nvme_free_space
```

Comment:
Is one of these `diff` calculations incorrect?

Reply:
This calculation is correct.